src/dodal/plans/verify_undulator_gap.py

```diff
@@ -0,0 +1,70 @@
+from dodal.devices.DCM import DCM
+from dodal.devices.undulator import Undulator, UndulatorGapAccess
+from dodal.log import LOGGER
+
+
+def _get_energy_to_distance_table(lookup_table_path: str) -> dict[float, float]:
+    energy_to_distance_table: dict[float, float] = {}
+    with open(lookup_table_path, mode="r") as table:
+        table_start = False
+        for line in table:
+            line = line.strip()
+            if line.startswith("Units"):
+                table_start = True
+                continue
+            if table_start:
+                value = line.split()
+                energy_to_distance_table[float(value[0])] = float(value[1])
+    return energy_to_distance_table
+
+
+def _get_closest_gap_to_dcm_energy(
+    dcm_energy: float, energy_to_distance_table: dict[float, float]
+) -> float:
+    table_energies = energy_to_distance_table.keys()
+    table_energy_value_to_use: float = 0
+    closest_distance_to_energy_from_table = dcm_energy
+    for energy in table_energies:
+        distance = abs(dcm_energy - energy)
+        if distance < closest_distance_to_energy_from_table:
+            closest_distance_to_energy_from_table = distance
+            table_energy_value_to_use = energy
+    return energy_to_distance_table[table_energy_value_to_use]
```

Comment:
Should: If you load into a `numpy` array as suggest above you will be able to do this in one or two lines using `numpy`